web/page0/src/test_apis.py

```diff
@@ -0,0 +1,330 @@
+"""APIs for test server.
+"""
+import flask
+from collections import defaultdict
+import json
+from time import sleep
+import random
+import string
+import asyncio
+
+def get_loop():
+    try:
+        loop = asyncio.get_running_loop()
+        if loop.is_running():
+            return loop
+        return None
+    except:
+        return None
+
+def bg_run(func, second):
+    async def task():
+        await asyncio.sleep(second)
+        func()
+
+    loop = get_loop()
+    if loop:
+        loop.create_task(task())
+    else:
+        asyncio.run(task())
+
+apis = flask.Blueprint("test_apis", __name__)
+
+mem = {"info": {"type": "cloud"}, "zones": defaultdict(list)}
+
+default_build_source = {
```

Comment:
If this variable is constant, it would be nice to have a name with all caps like DEFAULT_BUILD_SOURCE

Reply:
Done